fairworkflows/rdf_wrapper.py

```diff
@@ -48,7 +53,20 @@ def set_attribute(self, predicate, value):
         it (but throw a warning).
         """
         if self.get_attribute(predicate) is not None:
-            warnings.warn(f'A predicate {predicate} was already defined'
-                          f'overwriting {predicate} for {self.self_ref}')
+            warnings.warn(f'A predicate {predicate} was already defined\n'
+                          f'Overwriting {predicate} for {self.self_ref}')
             self._rdf.remove((self.self_ref, predicate, None))
         self._rdf.add((self.self_ref, predicate, value))
+        self._is_modified = True
```

Comment:
We should also anonymise upon setting an attribute right? i.e. in that case the RDF does not reflect the uri anymore. Also add a test for this behavior.

Reply:
For general addition of triples I suppose we should. Although note that here set_attribute only allows addition of triples of the form (this_step, predicate, object). So we'd be anonymising only for the case of (this_step, predicate, this_step) which is probably quite rare.

Reply:
Hmm. Actually a problem here might be that someone would legitimately want to add a triple that refers back to the URI of the step/workflow they are deriving this from. In that case they wouldn't want automatic anonymisation. In fact, I think that is a more likely triple a user might add than a triple where the subject and object are the same.

For that reason, I'd rather keep it as it is for now. Users can always use e.g. ```step.self_ref``` as the ```value``` when setting new attributes, if they really want the object to be the current step. It's imperfect, but I think not wanting anonymisation is more likely with use of this particular function.

Reply:
I agree, I was missing the point here that `self_ref` is actually referring to a blank node. So you are basically adding an anonomized RDF triple (with respect to the subject). And indeed if you want to add to RDF for example (this step, is derived from, uri) then you indeed do not want to anonomize automatically. Good point!